include/boost/geometry/algorithms/detail/distance/geometry_to_segment_or_box.hpp

```diff
@@ -203,25 +197,21 @@ class geometry_to_segment_or_box
         // consider all distances of the points in the geometry to the
         // segment or box
         comparable_return_type cd_min1(0);
-        point_iterator_type pit_min;
-        seg_or_box_const_iterator it_min1 = boost::const_begin(seg_or_box_points);
-        seg_or_box_const_iterator it_min2 = it_min1;
+        auto pit_min = points_begin(geometry);
+        auto it_min1 = boost::const_begin(seg_or_box_points);
```

Comment:
See above, we don't have that habit for iterators

Reply:
Defining it as `auto const` will prevent us from changing the iterator e.g incrementg it. If that's ok then it could be `const` like in case of end/last. Think about a pointer to const (`const_iterator`) vs const pointer (`auto const`).

Reply:
Yes, I changed it where @vissarion requested this. I completely agree now.

But specifically in this case, this iterator is meant to change. It is an iterator pointing to the minimum. So it cannot be made `const` here.